### PR TITLE
BX-4346 add option to skip preflight checks

### PR DESCRIPTION
**Pull request by rafaelvanoni**

### CODE REVIEW

**Review by aspin — APPROVED** [review]

File: bxserum/provider/http.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func (h *HTTPClient) SubmitCancelByClientOrderID(
 		return "", err
 	}
 
-	return h.signAndSubmit(order.Transaction)
+	return h.signAndSubmit(order.Transaction, true)
```
Comment (kdonthi) [inline bxserum/provider/http.go]: Do we want to give the option to enable preflight checks to the user for any transaction?

Comment (aspin) [reply bxserum/provider/http.go]: Actually yeah, I think we do. I'd be fine with either making it an RPCOpt to set on the whole provider or individually specified per endpoint

Comment (rafaelvanoni) [reply bxserum/provider/http.go]: From what I've seen the Cancel* endpoints always fail if these are enabled, so specifying by endpoint would probably be better than by provider. Will make these changes.

Comment (rafaelvanoni) [reply bxserum/provider/http.go]: Ok, pushed those changes, please take a look

**Review by aspin — APPROVED** [review]

**Pull request merged**